src/components/Slider/Slider.tsx

```diff
@@ -95,47 +69,71 @@ export const Slider = React.forwardRef(function Slider(
         tooltipDisplay,
         tooltipFormat,
     });
-
-    const stateModifiers: StateModifiers = React.useMemo<StateModifiers>(
-        () => ({
-            size,
-            error: validationState === 'invalid' && !disabled,
-            disabled,
-            'tooltip-display': innerState.tooltipDisplay,
-            rtl: direction === 'rtl',
-            'no-marks': Array.isArray(marks) ? marks.length === 0 : marks === 0,
-        }),
-        [direction, disabled, innerState.tooltipDisplay, size, validationState, marks],
+    const [innerValue, setValue] = useControlledState(
+        innerState.value,
+        innerState.defaultValue ?? min,
+        onUpdate,
     );
 
-    const handleRender = React.useMemo(
-        () =>
-            innerState.tooltipDisplay === 'off'
-                ? undefined
-                : (
-                      originHandle: HandleWithTooltipProps['originHandle'],
-                      originHandleProps: HandleWithTooltipProps['originHandleProps'],
-                  ) => (
-                      <HandleWithTooltip
-                          originHandle={originHandle}
-                          originHandleProps={originHandleProps}
-                          stateModifiers={stateModifiers}
-                          className={b('tooltip')}
-                          tooltipFormat={innerState.tooltipFormat}
-                      />
-                  ),
-        [innerState.tooltipDisplay, innerState.tooltipFormat, stateModifiers],
+    const handleReset = React.useCallback(
+        (v: number | [number, number]) => {
+            setValue(v);
+            onUpdateComplete?.(v);
+        },
+        [onUpdateComplete, setValue],
     );
 
+    const inputRef = useFormResetHandler({initialValue: innerValue, onReset: handleReset});
+
+    const stateModifiers: StateModifiers = {
+        size,
+        error: validationState === 'invalid' && !disabled,
+        disabled,
+        'tooltip-display': innerState.tooltipDisplay,
+        rtl: direction === 'rtl',
+        'no-marks': Array.isArray(marks) ? marks.length === 0 : marks === 0,
+    };
```

Comment:
why did you remove memoization? 
it can help prevent creating object on each render and could prevent rerenders inside components